core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -524,72 +489,105 @@ public static void deleteCancerStudyByCascade(int internalCancerStudyId) throws
      */
     public static void deleteCancerStudy(int internalCancerStudyId) throws DaoException {
+        String[] deleteStudyStatements = {
+                "DELETE FROM sample_cna_event WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM genetic_alteration WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM genetic_profile_samples WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM sample_profile WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM mutation WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM mutation_count WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM clinical_attribute_meta WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM clinical_event_data WHERE CLINICAL_EVENT_ID IN (SELECT CLINICAL_EVENT_ID FROM clinical_event WHERE PATIENT_ID IN (SELECT INTERNAL_ID FROM patient WHERE CANCER_STUDY_ID=?))",
+                "DELETE FROM clinical_event WHERE PATIENT_ID IN (SELECT INTERNAL_ID FROM patient WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM sample_list_list WHERE LIST_ID IN (SELECT LIST_ID FROM sample_list WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM clinical_sample WHERE INTERNAL_ID IN (SELECT INTERNAL_ID FROM sample WHERE PATIENT_ID IN (SELECT INTERNAL_ID FROM patient WHERE CANCER_STUDY_ID=?))",
+                "DELETE FROM sample WHERE PATIENT_ID IN (SELECT INTERNAL_ID FROM patient WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM clinical_patient WHERE INTERNAL_ID IN (SELECT INTERNAL_ID FROM patient WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM patient WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM copy_number_seg WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM copy_number_seg_file WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM sample_list WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM genetic_profile WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM gistic_to_gene WHERE GISTIC_ROI_ID IN (SELECT GISTIC_ROI_ID FROM gistic WHERE CANCER_STUDY_ID=?)",
+                "DELETE FROM gistic WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM mut_sig WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM protein_array_data WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM protein_array_cancer_study WHERE CANCER_STUDY_ID=?",
+                "DELETE FROM cancer_study WHERE CANCER_STUDY_ID=?;",
+                "DELETE FROM structural_variant WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)"
```

Comment:
Ok, I have moved the structural_variant purging to come before the genetic_profiles and the cancer_study.